### PR TITLE
motorcar:conditional should not apply to motorcycle and moped

### LINKED ISSUE #4355: "motorcar:conditional" should not restrict motorcycle
https://github.com/valhalla/valhalla/blob/master/src/mjolnir/pbfgraphparser.cc#L2390-L2393

Any particular reason to apply motorcar:conditional to motorcycle is a different kind?

### PR DESCRIPTION
**Pull request by nvcnvn**
# Issue

fixes #4355

## Tasklist

 - [ ] Add tests
 - [x] Add #fixes with the issue number that this PR addresses
 - [ ] Update the docs with any new request parameters or changes to behavior described
 - [x] Update the [changelog](CHANGELOG.md)
 - [ ] If you made changes to the lua files, update the [taginfo](taginfo.json) too.

### CODE REVIEW

File: src/mjolnir/pbfgraphparser.cc [hunk 1]
```diff
@@ -2387,10 +2387,12 @@ struct graph_callback : public OSMPBF::Callback {
         if (tokens.size() == 2 && tmp.size()) {
 
           uint16_t mode = 0;
-          if (boost::algorithm::starts_with(tag_.first, "motorcar:conditional") ||
-              boost::algorithm::starts_with(tag_.first, "motor_vehicle:conditional")) {
+          if (boost::algorithm::starts_with(tag_.first, "motor_vehicle:conditional")) {
```
Comment (nilsnolde) [inline src/mjolnir/pbfgraphparser.cc:2390]: we have to respect `motorcar` as well though. I'm not sure what would be the best thing here, as that usually implies in some way that motorcycle & motorscooter are affected as well. If it's a `kTimedDenied` it usually doesn't count for motorcycles and they're still allowed, but if it's a `kTimedAllowed` it should also allow motorcycles. so maybe it `motorcar` needs to distinguish the type? That's what https://wiki.openstreetmap.org/wiki/Key:motorcar suggests.

File: src/mjolnir/pbfgraphparser.cc [hunk 2]
```diff
@@ -2387,10 +2387,12 @@ struct graph_callback : public OSMPBF::Callback {
...
             mode = (kAutoAccess | kTruckAccess | kEmergencyAccess | kTaxiAccess | kBusAccess |
                     kHOVAccess | kMopedAccess | kMotorcycleAccess);
+          } else if (boost::algorithm::starts_with(tag_.first, "motorcar:conditional")) {
+            mode = (kAutoAccess | kTruckAccess | kEmergencyAccess | kTaxiAccess | kBusAccess |
+                    kHOVAccess);
```
Comment (nilsnolde) [inline src/mjolnir/pbfgraphparser.cc:2395]: Oh sorry, for some reason my eyes didn't see that you're doing that here.. Anyways, I think this is too broad. This is only valid for `kTimeDenied`. for `kTimeAllowed` we don't need to add anything I think:

```suggestion
          } else if (boost::algorithm::starts_with(tag_.first, "motorcar:conditional")) {
          if (type == kTimedDenie) {
             mode = (kAutoAccess | kTruckAccess | kEmergencyAccess | kTaxiAccess | kBusAccess |
                      kHOVAccess);
           }
```

Comment (nvcnvn) [reply src/mjolnir/pbfgraphparser.cc]: I will follow you suggestion. (I only know how to drive motorcycle LoL)

File: src/mjolnir/pbfgraphparser.cc (hunk 1 above)
Comment (nilsnolde) [reply src/mjolnir/pbfgraphparser.cc:2390]: @gknisely I personally only looked at the wiki definition. IMO it's pretty clear that both motorcar and motorcycle are a subset of motor_vehicle, but motorcycle can't be a subset of motorcar, it's clearly a car.. If any mapper doesn't think so, I personally don't care, I can only stomach a certain level of OSM craziness :smile:

File: src/mjolnir/pbfgraphparser.cc [hunk 3]
```diff
@@ -2387,10 +2387,16 @@ struct graph_callback : public OSMPBF::Callback {
         if (tokens.size() == 2 && tmp.size()) {
 
           uint16_t mode = 0;
-          if (boost::algorithm::starts_with(tag_.first, "motorcar:conditional") ||
-              boost::algorithm::starts_with(tag_.first, "motor_vehicle:conditional")) {
+          if (boost::algorithm::starts_with(tag_.first, "motor_vehicle:conditional")) {
             mode = (kAutoAccess | kTruckAccess | kEmergencyAccess | kTaxiAccess | kBusAccess |
                     kHOVAccess | kMopedAccess | kMotorcycleAccess);
+          } else if (boost::algorithm::starts_with(tag_.first, "motorcar:conditional")) {
+            if (type == kTimedAllowed) {
```
Comment (nilsnolde) [inline src/mjolnir/pbfgraphparser.cc:2394]: the build fails @nvcnvn 
```suggestion
            if (type == AccessType::kTimedAllowed) {
```

**Review by nilsnolde — DISMISSED** [review]

File: src/mjolnir/pbfgraphparser.cc (hunk 1 above)
Comment (gknisely) [reply src/mjolnir/pbfgraphparser.cc:2390]: fair enough

**Review by gknisely — DISMISSED** [review]

**Review by nilsnolde — APPROVED** [review]

**Comment by nilsnolde** [comment]
argh, now both arm & osx are failing with https://github.com/valhalla/valhalla/issues/4379

**Pull request merged**